static/js/components/dashboard/CourseDescription.js

```diff
@@ -68,13 +68,34 @@ export default class CourseDescription extends React.Component {
     return _.compact([dateMessage, additionalDetail]);
   }
 
-  renderViewCourseLink = (courseRun: CourseRun): React$Element<*>|null => (
-    (courseRun && courseRun.course_id) ?
-      <a href={`${EDX_LINK_BASE}${courseRun.course_id}`} target="_blank">
-        View on edX
-      </a> :
-      null
-  );
+  isCurrentOrPastEnrolled = (courseRun: CourseRun): boolean => {
+    if([STATUS_CURRENTLY_ENROLLED, STATUS_PASSED, STATUS_NOT_PASSED].includes(courseRun.status)) {
+      return true;
+    } else {
+      if ([STATUS_CAN_UPGRADE, STATUS_MISSED_DEADLINE].includes(courseRun.status)) {
+        let now = moment();
+        let courseStart = moment(courseRun.course_start_date);
+        return courseStart.isBefore(now);
+      } else {
+        return false;
+      }
+    }
+  };
+
+  renderViewCourseLink = (courseRun: CourseRun): React$Element<*>|null => {
+    if (!courseRun || !courseRun.course_id) {
+      return null;
+    }
+    let url = null;
+
+    if (this.isCurrentOrPastEnrolled(courseRun)) {
+      url = `${EDX_LINK_BASE}${courseRun.course_id}`;
```

Comment:
if `course_id` is null or undefined, `url` will not be blank, so it will result in a link to the edX base url. is that what we want? if not, we need to handle this case.

```javascript
if (courseRun.course_id && this.isCurrentOrPastEnrolled(courseRun)) {
  url = `${EDX_LINK_BASE}${courseRun.course_id}`;
// ...
```